pkg/cloud/cloud.go

```diff
@@ -101,7 +104,21 @@ type cloud struct {
 // It panics if session is invalid
 func NewCloud() (Cloud, error) {
 	sess := session.Must(session.NewSession(&aws.Config{}))
-	metadata, err := NewMetadataService(sess)
+	svc := ec2metadata.New(sess)
+	api, err := DefaultKubernetesAPIClient()
+
+	if err != nil && !isDriverBootedInECS() {
+		return nil, fmt.Errorf("could not instantiate k8s api client: %v", err)
```

Comment:
shouldn't error here, to maintain backwards compatibility the driver should tolerate k8s api being unavailable. it should only require kubernetes API to be available after it has determined the other sources are unavailable

Reply:
Have fixed this in latest version. Realised the error handling around MetadataProvider wasn't the best either so that's been changed, let me know if that fits your requirements.